### PR TITLE
add links to release notes, light refactor of landing page

### PR DESCRIPTION
**Pull request by techdocsmith**
Adds links to release notes from Druid landing/information page. Edit the landing page for style.
cc: @sthetland , @petermarshallio

This PR has:
- [x] been self-reviewed.

### CODE REVIEW

File: docs/design/index.md [hunk 1]
```diff
@@ -22,79 +22,74 @@ title: "Introduction to Apache Druid"
...
+is designed to run continuously without planned downtime for any reason. This is true for configuration changes and software
 updates.
-6. **Cloud-native, fault-tolerant architecture that won't lose data.** Once Druid has ingested your data, a copy is
-stored safely in [deep storage](architecture.md#deep-storage) (typically cloud storage, HDFS, or a shared filesystem).
-Your data can be recovered from deep storage even if every single Druid server fails. For more limited failures affecting
-just a few Druid servers, replication ensures that queries are still possible while the system recovers.
+6. **Cloud-native, fault-tolerant architecture that won't lose data.** After ingestion, Druid safely stores a copy of your data in [deep storage](architecture.md#deep-storage). Deep storage is typically cloud storage, HDFS, or a shared filesystem. You can recover your data from deep storage even in the unlikely case that all Druid servers fail. For a limited failure that affects only a few Druid servers, Druid uses replication to ensure that queries are still possible during system recovers.
```
Comment (sthetland) [inline docs/design/index.md:55]: I think "recovers" -> "recoveries", although the original seems to refer a little more clearly to "For a limited failure that affects..":

```suggestion
6. **Cloud-native, fault-tolerant architecture that won't lose data.** After ingestion, Druid safely stores a copy of your data in [deep storage](architecture.md#deep-storage). Deep storage is typically cloud storage, HDFS, or a shared filesystem. You can recover your data from deep storage even in the unlikely case that all Druid servers fail. For a limited failure that affects only a few Druid servers, replication ensures that queries are still possible during system recoveries.
```

File: docs/design/index.md [hunk 2]
```diff
@@ -22,79 +22,74 @@ title: "Introduction to Apache Druid"
...
 7. **Indexes for quick filtering.** Druid uses [Roaring](https://roaringbitmap.org/) or
-[CONCISE](https://arxiv.org/pdf/1004.0403) compressed bitmap indexes to create indexes that power fast filtering and
-searching across multiple columns.
-8. **Time-based partitioning.** Druid first partitions data by time, and can additionally partition based on other fields.
-This means time-based queries will only access the partitions that match the time range of the query. This leads to
-significant performance improvements for time-based data.
+[CONCISE](https://arxiv.org/pdf/1004.0403) compressed bitmap indexes to create indexes to enable fast filtering and searching across multiple columns.
+8. **Time-based partitioning.** Druid first partitions data by time. YOu can optionally implement additional partitioning based upon other fields.
```
Comment (sthetland) [inline docs/design/index.md:58]:
```suggestion
8. **Time-based partitioning.** Druid first partitions data by time. You can optionally implement additional partitioning based upon other fields.
```

File: docs/design/index.md [hunk 3]
```diff
@@ -22,79 +22,74 @@ title: "Introduction to Apache Druid"
...
+Time-based queries only access the partitions that match the time range of the query which leads to significant performance improvements.
 9. **Approximate algorithms.** Druid includes algorithms for approximate count-distinct, approximate ranking, and
 computation of approximate histograms and quantiles. These algorithms offer bounded memory usage and are often
 substantially faster than exact computations. For situations where accuracy is more important than speed, Druid also
 offers exact count-distinct and exact ranking.
 10. **Automatic summarization at ingest time.** Druid optionally supports data summarization at ingestion time. This
-summarization partially pre-aggregates your data, and can lead to big costs savings and performance boosts.
+summarization partially pre-aggregates your data potentially leading to significant cost savings and performance boosts.
```
Comment (sthetland) [inline docs/design/index.md:65]: The rewording is good, but it runs together without a comma, I think:
  
```suggestion
summarization partially pre-aggregates your data, potentially leading to significant cost savings and performance boosts.
```

File: docs/design/index.md [hunk 4]
```diff
@@ -22,79 +22,74 @@ title: "Introduction to Apache Druid"
...
-## When should I use Druid?
+## When to use Druid
 
-Druid is used by many companies of various sizes for many different use cases. Check out the
-[Powered by Apache Druid](/druid-powered) page
+Druid is used by many companies of various sizes for many different use cases. For more information see
+[Powered by Apache Druid](/druid-powered).
 
-Druid is likely a good choice if your use case fits a few of the following descriptors:
+Druid is likely a good choice if your use case has matches a few of the following:
```
Comment (sthetland) [inline docs/design/index.md:72]:
```suggestion
Druid is likely a good choice if your use case matches a few of the following:
```

**Review by sthetland — CHANGES_REQUESTED** [review]
A few typos. Otherwise looks good.

**Comment by techdocsmith** [comment]
Thanks @sthetland , committed those suggestions.

**Review by sthetland — APPROVED** [review]
Looks good!

**Pull request merged**